validation-test/stdlib/FloatingPointConversion.swift.gyb

```diff
@@ -0,0 +1,159 @@
+}
+
+%    else:
+
+/// Always-truncating conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we check if we fail and crash when an float would be truncated in conversion.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectNotEqual(input, resultConvertedBack)
+}
+
+/// Always-nil failable conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}FailableConversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we check if we return nil when a float would be truncated in conversion.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(exactly: input)
+  expectEqual(nil, result)
+}
+
+%    end
+
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=NaN") {
+  // Check that initializing with NaN from another type works.
+  let input = ${OtherFloat}.nan
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectTrue(input.isNaN)
+  expectTrue(resultConvertedBack.isNaN)
+}
+
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}Conversion/dest=NaN") {
+  // Check that initializing with NaN doesn't fail the initializer.
+  let input = ${OtherFloat}.nan
+  var result = ${Self}(exactly: input)
+  print(input)
+  print(result)
+  expectEqual(nil, result)
+}
+
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=inf") {
+  // Check that initializing with NaN from another type works.
+  let input = ${OtherFloat}.infinity
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectEqual(input, resultConvertedBack)
+}
+
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}Conversion/dest=inf") {
+  // Check that initializing with Inf from another type works.
+  let input = ${OtherFloat}.infinity
+  var result = ${Self}(exactly: input)
+  expectNotEqual(nil, result)
+}
+
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=-inf") {
+  // Check that initializing with NaN from another type works.
+  let input = ${OtherFloat}.infinity * -1
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectEqual(input, resultConvertedBack)
+}
+
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}Conversion/dest=-inf") {
+  // Check that initializing with -Inf from another type works.
+  let input = ${OtherFloat}.infinity * -1
+  var result = ${Self}(exactly: input)
+  expectNotEqual(nil, result)
+}
+
+%     if OtherFloat == 'Float80':
+#endif
+%     end
+
+%   end # for in floatNameToSignificandBits (Other)
+
+%     if Self == 'Float80':
+#endif
+%     end
+
+% end # for in floatNameToSignificandBits (Self)
+
+runAllTests()
```

Comment:
I wonder if there is a way to make the tests table-driven and test more interesting values like we do in `test/1_stdlib/FloatingPoint.swift.gyb` (`let interestingValues`).